boards/adafruit-macropad/src/macros.rs

```diff
@@ -0,0 +1,209 @@
+            // (16_000_000u32 as embedded_time::rate::Extensions).Hz(),
+            &$crate::MODE_0,
+        );
+
+        let mut oled_reset = $pins.oled_reset.into_push_pull_output();
+
+        let mut disp: $crate::GraphicsMode<_> = sh1106::Builder::new()
+            .connect_spi(
+                spi1,
+                $pins.oled_dc.into_push_pull_output(),
+                $pins.oled_cs.into_push_pull_output(),
+            )
+            .into();
+
+        disp.reset(&mut oled_reset, &mut $delay).unwrap();
+
+        disp.init().unwrap();
+        disp
+    }};
+}
+
+#[macro_export]
+macro_rules! macropad_pins {
+    ($pac:expr, $sio:expr) => {
+        $crate::Pins::new(
+            $pac.IO_BANK0,
+            $pac.PADS_BANK0,
+            $sio.gpio_bank0,
+            &mut $pac.RESETS,
+        )
+    };
+}
+
+#[macro_export]
+macro_rules! macropad_clocks {
+    ($pac:expr, $watchdog:expr) => {
+        $crate::init_clocks_and_plls(
+            $crate::XOSC_CRYSTAL_FREQ,
+            $pac.XOSC,
+            $pac.CLOCKS,
+            $pac.PLL_SYS,
+            $pac.PLL_USB,
+            &mut $pac.RESETS,
+            &mut $watchdog,
+        )
+        .ok()
+        .unwrap()
+    };
+}
+
+#[macro_export]
+macro_rules! macropad_rotary_encoder {
+    ($pins:expr) => {
+        $crate::Rotary::new(
+            $pins.encoder_rota.into_pull_up_input(),
+            $pins.encoder_rotb.into_pull_up_input(),
+        )
+    };
+}
+
+#[macro_export]
+macro_rules! macropad_keypad {
+    ($pins:expr) => {
+        $crate::KeysTwelve {
+            key1: $pins.key1.into_pull_up_input(),
+            key2: $pins.key2.into_pull_up_input(),
+            key3: $pins.key3.into_pull_up_input(),
+            key4: $pins.key4.into_pull_up_input(),
+            key5: $pins.key5.into_pull_up_input(),
+            key6: $pins.key6.into_pull_up_input(),
+            key7: $pins.key7.into_pull_up_input(),
+            key8: $pins.key8.into_pull_up_input(),
+            key9: $pins.key9.into_pull_up_input(),
+            key10: $pins.key10.into_pull_up_input(),
+            key11: $pins.key11.into_pull_up_input(),
+            key12: $pins.key12.into_pull_up_input(),
+        }
+    };
+}
+
+pub struct KeysTwelve {
+    pub key1: Pin<Gpio1, PullUpInput>,
+    pub key2: Pin<Gpio2, PullUpInput>,
+    pub key3: Pin<Gpio3, PullUpInput>,
+    pub key4: Pin<Gpio4, PullUpInput>,
+    pub key5: Pin<Gpio5, PullUpInput>,
+    pub key6: Pin<Gpio6, PullUpInput>,
+    pub key7: Pin<Gpio7, PullUpInput>,
+    pub key8: Pin<Gpio8, PullUpInput>,
+    pub key9: Pin<Gpio9, PullUpInput>,
+    pub key10: Pin<Gpio10, PullUpInput>,
+    pub key11: Pin<Gpio11, PullUpInput>,
+    pub key12: Pin<Gpio12, PullUpInput>,
+}
+
+impl KeysTwelve {
+    pub fn get_0based(&self, idx: i8) -> Option<&dyn InputPin<Error = Infallible>> {
+        self.get_1based(1 + idx)
+    }
+
+    pub fn get_1based(&self, idx: i8) -> Option<&dyn InputPin<Error = Infallible>> {
```

Comment:
Do you expect people to use both of these interfaces?
If you don't, it would best to pick the one you think is most useful, add a doc comment indicating how it's supposed to be used, and leave it with the name `get`